ntoskrnl/config/ntapi.c

```diff
@@ -1454,40 +1315,416 @@ NtLockRegistryKey(IN HANDLE KeyHandle)
 
 NTSTATUS
 NTAPI
-NtNotifyChangeMultipleKeys(IN HANDLE MasterKeyHandle,
-                           IN ULONG Count,
-                           IN POBJECT_ATTRIBUTES SlaveObjects,
-                           IN HANDLE Event,
-                           IN PIO_APC_ROUTINE ApcRoutine OPTIONAL,
-                           IN PVOID ApcContext OPTIONAL,
-                           OUT PIO_STATUS_BLOCK IoStatusBlock,
-                           IN ULONG CompletionFilter,
-                           IN BOOLEAN WatchTree,
-                           OUT PVOID Buffer,
-                           IN ULONG Length,
-                           IN BOOLEAN Asynchronous)
+NtNotifyChangeMultipleKeys(
+    IN HANDLE MasterKeyHandle,
+    IN ULONG Count,
+    IN POBJECT_ATTRIBUTES SlaveObjects,
+    IN HANDLE Event,
+    IN PIO_APC_ROUTINE ApcRoutine OPTIONAL,
+    IN PVOID ApcContext OPTIONAL,
+    OUT PIO_STATUS_BLOCK IoStatusBlock,
+    IN ULONG CompletionFilter,
+    IN BOOLEAN WatchTree,
+    OUT PVOID Buffer,
+    IN ULONG Length,
+    IN BOOLEAN Asynchronous)
 {
-    UNIMPLEMENTED_ONCE;
-    return STATUS_NOT_IMPLEMENTED;
+    NTSTATUS Status;
+    KPROCESSOR_MODE PreviousMode = ExGetPreviousMode();
+
+    /* Registry Key */
+    PCM_KEY_BODY KeyObject = NULL;
+
+    /* Notification state blocks */
+    PCM_NOTIFY_BLOCK NotifyBlock = NULL;
+    PCM_POST_BLOCK PostBlock = NULL;
+
+    /* Event Objects */
+    HANDLE LocalEventHandle = NULL;
+    PKEVENT LocalEventObject = NULL;
+
+    /* Subordinate objects */
+    OBJECT_ATTRIBUTES *LocalSubObjects = NULL;
+    UNICODE_STRING *SubNames = NULL;
+    PCM_KEY_BODY *LocalSubObjectsKeyBody = NULL;
+    PCM_NOTIFY_BLOCK *SubNotifyBlocks = NULL;
+
+    PAGED_CODE();
+
+    DPRINT(
+        "NtNotifyChangeMultipleKeys(MasterKeyHandle=%p, Subordinate[%d]=%p, Event=%p, ApcRoutine=%p, ApcContext=%p, "
+        "IoStatusBlock=%p, CompletionFilter=%d, WatchTree=%d, Buffer[%d]=%p, Asynchronous=%d)\n",
+        MasterKeyHandle, Count, SlaveObjects, Event, ApcRoutine, ApcContext, IoStatusBlock, CompletionFilter, WatchTree,
+        Buffer, Length, Asynchronous);
+
+    /* Validate flags */
+    if ((CompletionFilter & REG_LEGAL_CHANGE_FILTER) != CompletionFilter)
+        return STATUS_INVALID_PARAMETER;
+
+    /* For asynchronous calls, either Event or APC routine should be provided */
+    if (Asynchronous && (Event == NULL && ApcRoutine == NULL))
+        return STATUS_INVALID_PARAMETER;
+
+    /* ApcContext can't be null when ApcRoutine is not */
+    if (PreviousMode != KernelMode && ApcRoutine != NULL && ApcContext == NULL)
+        return STATUS_INVALID_PARAMETER;
+
+    /* IoStatusBlock is required */
+    if (IoStatusBlock == NULL)
+        return STATUS_INVALID_PARAMETER;
+
+    /* Windows doesn't support more than one subordinate */
+    if (Count != 0 && Count != 1)
```

Comment:
`Count` is a ULONG so if the implementation only supports one subordinate you could just do `Count > 1` if it's above the supported subordinate.